File: Sorting/Insertion-Sort.py
def InsertionSort(arr):
    for i in range(1,len(arr)):
        j = i -1
        key = arr[i]

        # Moving elements of arr[0...i-1], which are greater than key
        # to one position ahead of their current position 
        while(j >=0 and arr[j]>key):
            arr[j+1] = arr[j]
            j = j-1
        #inserting key at it's correct position in sorted
        arr[j+1] = key

#Taking the input
array = list(map(int,input("Enter your array (as space separated integers) : ").split()))

#Passing array to the function
InsertionSort(array)

print(array)

File: C++/Sorting/Insertion-Sort.cpp
#include <iostream>

void Insertion_Sort(int arr[], int n);
void swap(int* a, int* b);

int main(){

    int arr[50], n;

    std::cout<< "Enter the size of your array : " ;
    std::cin >> n;

    std::cout<< "Enter the elements of your array : " ;
    for(int i=0; i<n; ++i)
        std::cin >>arr[i];

    Insertion_Sort(arr,n);

    std::cout<< "Sorted array : " << std::endl;
    for(int i=0; i<n; ++i)
        std::cout << arr[i] << ",";

    return 0;    
}

void Insertion_Sort(int arr[], int n){

    for (int i=1; i<n; i++){
        int temp = arr[i];
        int j = i-1;

        while(j>=0 && arr[j] > temp){
            arr[j+1] = arr[j];
            j --;
        }
        arr[j+1] = temp;
    }
}

void swap(int* a, int* b){

    int temp;

    temp = *a;
    *a = *b;
    *b = temp;
}
